phones/models.py

```diff
@@ -231,12 +233,20 @@ def save(self, *args, **kwargs):
         # as realphone
         self.country_code = realphone.country_code.upper()
 
+        # Add US numbers to the Relay messaging service, so it goes into our
+        # US A2P 10DLC campaign
         if self.country_code == "US":
-            # Also add this number to the Relay messaging service, so it goes
-            # into our US A2P 10DLC campaign
-            client.messaging.v1.services(
-                settings.TWILIO_MESSAGING_SERVICE_SID
-            ).phone_numbers.create(phone_number_sid=twilio_incoming_number.sid)
+            try:
+                client.messaging.v1.services(
+                    settings.TWILIO_MESSAGING_SERVICE_SID
+                ).phone_numbers.create(phone_number_sid=twilio_incoming_number.sid)
+            except TwilioRestException as err:
+                if err.status == 409 and err.code == 21710:
+                    # Ignore "Phone Number is already in the Messaging Service"
+                    # https://www.twilio.com/docs/api/errors/21710
+                    pass
+                else:
+                    raise
```

Comment:
While this is happening because we are trying to allow multiple Relay Number for the same Real Number, but this could happen if we/Twilio accidentally serve a Relay Number that is already used (possible due to race condition etc?). Should we check here again that the Relay Number is correctly assigned to this user/Real Phone?

Reply:
Yes, this is the only code preventing two users from claiming the same relay number 🤦 
And, it won't work in Canada.